enterprise/views.py

```diff
@@ -887,73 +976,19 @@ def post(self, request):
-                        else 'professional' if 'professional' in course_modes \
-                        else 'no-id-professional' if 'no-id-professional' in course_modes \
-                        else 'audit'
-                    try:
-                        enrollment_api_client.enroll_user_in_course(
-                            request.user.username,
-                            course_id,
-                            course_mode
-                        )
-                        LOGGER.info(
-                            'Created LMS enrollment for User {user} in Course {course_id} '
-                            'with License {license_uuid} in Course Mode {course_mode}.'.format(
-                                user=request.user.username,
-                                course_id=course_id,
-                                license_uuid=license_uuid,
-                                course_mode=course_mode
-                            )
-                        )
-                    except Exception as exc:    # pylint: disable=broad-except
-                        LOGGER.error(
-                            'Unable to create an LMS enrollment from the DSC view: {exc}'.format(
-                                exc=exc
-                            )
-                        )
-                        return redirect(failure_url)
-                try:
-                    self.create_enterprise_course_enrollment(request, consent_record, course_id, license_uuid)
-                except IntegrityError:
-                    error_code = 'ENTGDS009'
-                    log_message = (
-                        '[Enterprise DSC API] IntegrityError while creating EnterpriseCourseEnrollment.'
-                        'Course: {course_id}, '
-                        'Program: {program_uuid}, '
-                        'EnterpriseCustomer: {enterprise_customer_uuid}, '
-                        'User: {user_id}, '
-                        'License UUID: {license_uuid}, '
-                        'ErrorCode: {error_code}'.format(
-                            course_id=course_id,
-                            program_uuid=program_uuid,
-                            enterprise_customer_uuid=enterprise_uuid,
-                            user_id=request.user.id,
-                            license_uuid=license_uuid,
-                            error_code=error_code,
-                        )
-                    )
-                    LOGGER.exception(log_message)
+            try:
+                self._enroll_leaner_in_course(
+                    request=request,
+                    enterprise_customer=enterprise_customer,
+                    course_id=course_id,
+                    program_uuid=program_uuid,
+                    license_uuid=license_uuid)
+            except Exception:  # pylint: disable=broad-except
+                pass
             consent_record.granted = consent_provided
             consent_record.save()
```

Comment:
The consent record update can go within the try block and we should be redirecting to the failure_url when we catch the exception

Reply:
Well all the current unit tests seem to document that that's not the expected behavior and I don't think we should be changing up the expected behavior in a section of the code not being actively worked on.

If you _really_ want me to change up the behavior then I'm going to need to fix a lot of unit tests and manually check those enrollment cases for regressions and given the DSC screen doesn't currently work on my local environment we'd probably need to repoint this ticket and throw on a few more points

Reply:
the point of the exceptions are to know when to redirect to the `failure_url` in the method you are calling `_enroll_leaner_in_course` from. as of right now you don't redirect on failure the same way as the original logic did

In the case of the `InvalidKeyError` and `IntegrityError` you should not be re-raising the exception because the original logic didn't redirect to `failure_url` in those cases. For the general `Exception` catch block in `_enroll_leaner_in_course`, where a redirect to the `failure_url` was originally triggered on failure, you should be re-raising the exception so that you know to redirect from the calling method

Reply:
I'm happy to hop on a call if this needs more explanation, I know communicating solely through text can be difficult